### PR TITLE
Add Debugging DOCS

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

I realized that there are a lot of tips that i have for debugging that could be useful. This doc is designed to be a jumping off point. I'm sure there are more strategies that should be useful.

### Test Plan

@irfanhudda thank you so much for the help with `remark` I feel much more confident about these docs

### CODE REVIEW

File: docs/debugging.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+### Client
+
+Perhaps, the most interesting question you can ask about the Debugge is how does it work?
+This questions often takes the form of, "how is a breakpoint added?" or "what happens when the debugger pauses?".
+
+To answer these questions, you need to look at the Debugger's client, which talks to the server over a websocket.
+
+The client commarnds are available on the console at `client` for testing purposes:
```
Comment (irfanhudda) [inline docs/debugging.md]: Typo here `commarnds`

File: docs/debugging.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+It's nice to look at the unit tests to see how an action *should* be called. If you don't see the use case you're looking for, add a unit test to try it out. Once you get it working you can PR the new unit test and everyone benefits :)
+
+### Reducers
+
+It's really common to want to know what data the Debugger has.
+For instance, what does a breakpoint look like?
+In Redux, the reducers are like the application database and it's tremendously useful to be able to inspect them.
+
+It's easy to view the current Debugger state in the console with
+the `appStore` global. Because the data is immutable, you'll need to request the state as a pojo.
```
Comment (nchevobbe) [inline docs/debugging.md]: don't know if we could have `<abbr>` in markdown, but here `<abbr title="Plain Old Javascript Object">pojo</abbr>` would be nice

File: docs/debugging.md (hunk 1 above)
Comment (nchevobbe) [inline docs/debugging.md]: s/commarnds/commands

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -155,21 +170,12 @@ Go to [Pull Requests](./docs/pull-requests.md) to learn about:
...
-* [Configs](./docs/local-development.md#configs)
-* [Hot Reloading](./docs/local-development.md#hot-reloading-fire)
-* [Themes](./docs/local-development.md#themes)
-* [Internationalization](./docs/local-development.md#internationalization)
-* [Prefs](./docs/local-development.md#prefs)
-* [Flow](./docs/local-development.md#flow)
-* [Logging](./docs/local-development.md#logging)
-* [Testing](./docs/local-development.md#testing)
-* [Linting](./docs/local-development.md#linting)
+### Debugging
```
Comment (nchevobbe) [inline CONTRIBUTING.md]: Could there be a link to the debugging.md (without anchor) here ?

File: docs/debugging.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+{"from":"server2.conn7.child1/27","type":"newSource","source":{"actor":"server2.conn7.child1/29","generatedUrl":null,"url":"http://localhost:7999/mutating.html","isBlackBoxed":false,"isPrettyPrinted":false,"isSourceMapped":false,"sourceMapURL":null,"introductionUrl":null,"introductionType":null}}
+{"sources":[{"actor":"server2.conn7.child1/29","generatedUrl":null,"url":"http://localhost:7999/mutating.html","isBlackBoxed":false,"isPrettyPrinted":false,"isSourceMapped":false,"sourceMapURL":null,"introductionUrl":null,"introductionType":null}],"from":"server2.conn7.child1/27"}
+{"to":"server2.conn7.child1/29","type":"source"}
+```
+
+### Communication
+
+Lastly, it's worth mentioning that Developer Tools is an *advanced* subject
+you're actually inquiring into how JS works. It's awesome that you're curious and there
+are lots of people in our [slack] channel who are learning along side of you.
```
Comment (irfanhudda) [inline docs/debugging.md]: Minor change `alongside`

File: docs/debugging.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+The best thing about React and Redux is that it simplifies the development workflow.
+Generally speaking, code belongs in one of four buckets: component, action, reducer, or client.
+
+There are different strategies for debugging each bucket, which we'll outline below.
+
+### Components
+
+The first thing you want to find the component to work on.
+The react devtools extension is useful for find the correct component.
```
Comment (bomsy) [inline docs/debugging.md]: typo: for finding

File: docs/debugging.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+You can also test out a selector in the console with the `getGlobalsForTesting` helper:
+
+```js
+getGlobalsForTesting().selectors.getBreakpoints(appStore.getState())
+```
+
+### Client
+
+Perhaps, the most interesting question you can ask about the Debugge is how does it work?
```
Comment (princiya) [inline docs/debugging.md]: Typo `Debugger`

File: docs/debugging.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+This questions often takes the form of, "how is a breakpoint added?" or "what happens when the debugger pauses?".
```
Comment (princiya) [inline docs/debugging.md]: Grammar nazi ;) `question`

**Pull request merged**